### DISCUSSION TITLE
Query Builder filter by custom date fields does not work

### DISCUSSION THREAD

**Discussion by jhorii** [statamic/cms#5028] [Troubleshooting] [answered]
I am creating a custom tag that is to query entries and filter by exhibit location and start and end dates.  Fields are "exhibit_location", "exhibit_start_date", "exhibit_end_date".  I need this custom tag to return a true if there are active exhibits and return false if none.  From my page I will then display some default content if no active exhibits.  But I haven't got there yet because I cannot seem to get this query to work correctly.

The start and end date fields are standard default date fields with nothing changed as far as basic settings.  It is date only, no time or date range.

From what I can tell it's not actually comparing dates but possibly strings.  For simplicity I'm not going to include the "exhibit_end_date" in my example. In my testing, I have two entries in the Exhibitions collection
- title: Entry 1, exhibit_start_date: '2021-10-01'
- title: Entry 2, exhibit_start_date: '2022-01-07'

Here is my custom tag:
```
<?php

namespace App\Tags;

use Statamic\Tags\Tags;
use Statamic\Facades\Entry;
use Illuminate\Support\Carbon;

class Hsnc extends Tags
{
    /**
     * The {{ hsnc }} tag.
     *
     * @return string|array
     */
    public function index()
    {
        //
    }

    /**
     * The {{ hsnc:active_exhibits }} tag.
     * 
     *  What I intend for this to do is query for any active exhibits and get the count.  If the count is greater than 0 then return true
     *  But I haven't done that here yet because I can't get the query to work  properly.  It's currently setup to help me troubleshoot.
     */
    public function activeExhibits()
    {        
        $results = Entry::query()
        ->where('collection', 'exhibitions')
        ->where('exhibit_locations', $this->params->get('location'))
        ->where('exhibit_start_date', '<=', Carbon::parse('today'))
        ->get();

        return $results;
    }
}
```
And here is my tag on my page where I'm displaying the results, this is currently just to help troubleshoot:
```
{{ hsnc:active_exhibits location="main_gallery" }}                         
       {{ title }} : {{ exhibit_start_date }} 
{{ /hsnc:active_exhibits }}
```

The results should only return one entry, which is "Entry 1".  Today's current date is January 6, 2022.  But the results continually return both entries.  Even if I change the start date for both to be in the future, it still returns both.  I've tested returning the Carbon::parse('today') value, and it's correct as well. 

I've read all the documentation on it that I can and I believe I've followed it correctly.  For example: https://statamic.dev/repositories/entry-repository#querying

I was about 5 versions back, so I updated to 3.2.29 to make sure there wasn't a bug fix, but the issue is still there.

Any help here is appreciated.

**Comment by ryanmitchell** [comment, marked as answer]
Unfortunately I don't think date equivalence in query builders works as yet as it does a string comparison. I've an open PR which adds this function:

https://github.com/statamic/cms/pull/4753

**Reply by jhorii** [reply]
Thank you!  This will be extremely useful for my project.
